### ISSUE TITLE
Error opening SHP file with "Point (PointZ)"

### ISSUE THREAD

**Issue by ruslo** [CloudCompare/CloudCompare#1612]
The bug is similar to https://github.com/CloudCompare/CloudCompare/issues/1586

Error while trying to open SHP file:
```
[17:44:27] [SHP] Shape 184549376 has an invalid shape code (1088169507)
```

ZIP archive with shapefiles to reproduce: [points.zip](https://github.com/CloudCompare/CloudCompare/files/8283521/points.zip)

 - CC Version (or git hash/tag): 541800c9
 - OS & Version: Ubuntu 18.04
 - Qt Version (if compiling): 5.12.9

**Comment by dgirardeau** [comment]
I can reproduce the error on Windows. But before saying it's a bug, have you been able to load this file in another tool? How was it generated?

**Comment by ruslo** [comment]
> have you been able to load this file in another tool?

Yes, the file is opening fine in [QGIS](https://github.com/qgis/QGIS).

> How was it generated?

I'm calling GDAL API to generate the file, using `OGR_G_CreateGeometry(wkbPoint25D)` and friends.

**Comment by dgirardeau** [comment]
Ok, so I guessed it was another shape type for which the 'measurement' is optional... (we has to fix other shape types similarly in the past).

I fixed the code: 6679990b674d76eb62d81e8ead76977cfa90a951
